modules/xfem/include/auxkernels/XFEMMarkerAux.h

```diff
@@ -0,0 +1,47 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef XFEMMARKERAUX_H
+#define XFEMMARKERAUX_H
+
+#include "AuxKernel.h"
+
+class XFEM;
+
+/**
+ * Coupled auxiliary value
+ */
+class XFEMMarkerAux : public AuxKernel
+{
+public:
+
+  /**
+   * Factory constructor, takes parameters so that all derived classes can be built using the same
+   * constructor.
+   */
+  XFEMMarkerAux(const InputParameters & parameters);
+
+  virtual ~XFEMMarkerAux() {}
+
+protected:
+  virtual Real computeValue();
+
+private:
+  XFEM *_xfem;
```

Comment:
No.
http://stackoverflow.com/questions/1143262/what-is-the-difference-between-const-int-const-int-const-and-int-const
The member may have to be writable, but the object it points to probably
not.

On Mon, Feb 22, 2016 at 4:10 PM Ben Spencer <notifications@github.com>
wrote:

> In modules/xfem/include/auxkernels/XFEMMarkerAux.h
> <https://github.com/idaholab/moose/pull/6321#discussion_r53710067>:
>
> > +{
> > +public:
> > +
> > +  /**
> > +   * Factory constructor, takes parameters so that all derived classes can be built using the same
> > +   * constructor.
> > +   */
> > +  XFEMMarkerAux(const InputParameters & parameters);
> > +
> > +  virtual ~XFEMMarkerAux() {}
> > +
> > +protected:
> > +  virtual Real computeValue();
> > +
> > +private:
> > +  XFEM *_xfem;
>
> We can't set it in the initialization list, so yes, it does need to be
> writeable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6321/files#r53710067>.
>